Dangerfile

```diff
@@ -1,37 +1,49 @@
 # frozen_string_literal: true
 
-def release_branch?
-  danger.github.branch_for_base.start_with?('release/') || danger.github.branch_for_base.start_with?('hotfix/')
-end
+github.dismiss_out_of_range_messages
 
-def main_branch?
-  danger.github.branch_for_base == 'trunk'
-end
+# `files: []` forces rubocop to scan all files, not just the ones modified in the PR
+rubocop.lint(files: [], force_exclusion: true, inline_comment: true, fail_on_inline_comment: true, include_cop_names: true)
+
+manifest_pr_checker.check_all_manifest_lock_updated
+
+podfile_checker.check_podfile_does_not_have_branch_references
 
-def wip_feature?
-  has_wip_label = github.pr_labels.any? { |label| label.include?('WIP') }
-  has_wip_title = github.pr_title.include?('WIP')
+ios_release_checker.check_core_data_model_changed
+ios_release_checker.check_release_notes_and_app_store_strings
 
-  has_wip_label || has_wip_title
+# skip remaining checks if we're during the release process
+if github.pr_labels.include?('Releases')
+  message('This PR has the `Releases` label: some checks will be skipped.')
+  return
 end
 
-return if github.pr_labels.include?('Releases')
+common_release_checker.check_internal_release_notes_changed(report_type: :message)
 
-github.dismiss_out_of_range_messages
+ios_release_checker.check_modified_translations_on_release_branch
 
-manifest_pr_checker.check_all_manifest_lock_updated
+tracks_checker.check_tracks_changes(
+  tracks_files: [
+    'WooAnalyticsStat.swift'
+  ],
+  tracks_usage_matchers: [
+    /AnalyticsTracker\.track/
+  ],
+  tracks_label: 'Tracks'
+)
+
+view_changes_checker.check
+
+pr_size_checker.check_diff_size(max_size: 500)
+
+# skip remaining checks if we have a Draft PR
+return if github.pr_draft?
 
 labels_checker.check(
   do_not_merge_labels: ['status: do not merge'],
   required_labels: [//],
   required_labels_error: 'PR requires at least one label.'
 )
 
-view_changes_need_screenshots.view_changes_need_screenshots
-
-pr_size_checker.check_diff_size
-
 # skip check for draft PRs and for WIP features unless the PR is against the main branch or release branch
-milestone_checker.check_milestone_due_date(days_before_due: 2) unless github.pr_draft? || (wip_feature? && !(release_branch? || main_branch?))
-
-rubocop.lint(inline_comment: true, fail_on_inline_comment: true, include_cop_names: true)
+milestone_checker.check_milestone_due_date(days_before_due: 2) unless github_utils.wip_feature? && !(github_utils.release_branch? || github_utils.main_branch?)
```

Comment:
A nitpick for your consideration. 

```suggestion
milestone_checker.check_milestone_due_date(days_before_due: 2) unless github_utils.wip_feature? && !github_utils.release_branch? && !github_utils.main_branch?
```

Or, here's another option. What we want to do is check the milestone if we are targeting the release branch or the main branch and we are not on a WIP feature, so

```suggestion
milestone_checker.check_milestone_due_date(days_before_due: 2) if (github_utils.main_branch? || github_utils.release_branch?) && !github_utils.wip_feature?
```